django/template/base.py

```diff
@@ -927,6 +945,27 @@ def render_annotated(self, context):
                 e.template_debug = context.template.get_exception_info(e, self.token)
             raise
 
+    def stream(self, context):
+        """
+        Return a string generator rendering the node.
+        """
+        yield self.render(context)
+
+    def stream_annotated(self, context):
+        """
+        Render the node. If debug is True and an exception occurs during
+        rendering, the exception is annotated with contextual line information
+        where it occurred in the template. For internal usage this method is
+        preferred over using the render method directly.
+        """
+        try:
+            for chunk in self.stream(context):
+                yield chunk
+        except Exception as e:
+            if context.template.engine.debug and not hasattr(e, 'template_debug'):
+                e.template_debug = context.template.get_exception_info(e, self.token)
+            raise
```

Comment:
Could we maybe put this logic in a context manager or decorator? It'd be nice not to duplicate it from ``render_annotated``. Quick example:

```python
from contextlib import contextmanager

@contextmanager
def annotate_exception(context, token):
    try: 
        yield
    except Exception as e:
        if context.template.engine.debug and not hasattr(e, 'template_debug'):
            e.template_debug = context.template.get_exception_info(e, token)
        raise

def stream_annotated(self, context):
    with annotate_exception(context, self.token):
        for chunk in self.stream(context):
            yield chunk
```